src/library_fs.js

```diff
@@ -1544,6 +1544,17 @@ mergeInto(LibraryManager.library, {
       var path = PATH.join2(typeof parent === 'string' ? parent : FS.getPath(parent), name);
       return FS.symlink(target, path);
     },
+    ensureFolder: function(path, canRead, canWrite) {
```

Comment:
`ensureFolder` sounds a bit vague. Perhaps this would be better called `createDirectoryTree`? Also, what if there is a file with a conflicting name present? The function does not really ensure in the sense that it would return success or failure, or throw on error, but it will silently fail if the path is blocked by a file? Would we prefer to return success or failure, or throw on error?